message_ix/tools/add_year/__init__.py

```diff
@@ -409,37 +437,44 @@ def add_year_par(
         )
         sc_new.commit("New parameter initiated!")
 
-    if node_col:
-        par_old = sc_ref.par(parname, {node_col[0]: reg_list})
-        par_new = sc_new.par(parname, {node_col[0]: reg_list})
-        sort_order = [
+    par_old = (
```

Comment:
This I don't understand: we currently have:
```python
    par_old = (
        sc_ref.par(parname, {node_col[0]: reg_list})
        if node_col
        else sc_ref.par(parname)
    )
```
And the targeted `Scenario` below is `sc_new`, so different. 
If I understand correctly, we want `par_old` and `par_new` to be `DataFrames`, so setting them to `None` is not really an option. We could of course catch if they are `None` and then call `sc_ref.par(parname)` instead, but I don't see the difference to what we already have.

Reply:
Ah, sorry, I used the wrong left-hand side:
```python
par_old = sc_ref.par(parname, filters={node_col[0]: reg_list} if node_col else None)
```
Note that the entire ternary expression `{…} if node_col else None` only determines the value passed to the `filters=…` keyword argument; `par_old` is the return value of `Scenario.par()`, always a data frame though possibly empty.

The general tactic is "use the ternary expression for only the affected argument, instead of repeating the entire function call in the first and last parts of the expression".

Reply:
Thanks for the explanation, I see now :)